### PR TITLE
sci-astronomy/siril: add 1.2.0_rc1

### PR DESCRIPTION
**Pull request by hamarituc**

### CODE REVIEW

**Comment by ceamac** [comment]
Does not compile with gcc-13 (default on ~amd64)

```
FAILED: src/libsiril.a.p/core_exif.cpp.o 
x86_64-pc-linux-gnu-g++ -Isrc/libsiril.a.p -Isrc -I../siril-1.2.0-rc1/src -Isubprojects/htmesh/__CMake_build -I../siril-1.2.0-rc1/subprojects/htmesh/__CMake_build -Isubprojects/htmesh -I../siril-1.2.0-rc1/subprojects/htmesh -Isubprojects/kplot -I../siril-1.2.0-rc1/subprojects/kplot -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/lib64/libffi/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/opencv4 -I/usr/include/json-glib-1.0 -fdiagnostics-color=always -Wall -Winvalid-pch -std=c++14 -O2 -march=znver2 -pipe -frecord-gcc-switches -frecord-gcc-switches -fPIC -fopenmp -pthread -DHAVE_CONFIG_H -fcommon -fno-math-errno -D_FILE_OFFSET_BITS=64 -DHAVE_LIBRTPROCESS -DHAVE_FFTW3F_OMP -MD -MQ src/libsiril.a.p/core_exif.cpp.o -MF src/libsiril.a.p/core_exif.cpp.o.d -o src/libsiril.a.p/core_exif.cpp.o -c ../siril-1.2.0-rc1/src/core/exif.cpp
In file included from ../siril-1.2.0-rc1/src/core/exif.cpp:49:
../siril-1.2.0-rc1/src/core/exif.h:27:48: error: ‘uint8_t’ has not been declared
   27 | int siril_get_thumbnail_exiv(const char *path, uint8_t **buffer, size_t *size, char **mime_type);
      |                                                ^~~~~~~
../siril-1.2.0-rc1/src/core/exif.cpp:78:48: error: ‘uint8_t’ has not been declared
   78 | int siril_get_thumbnail_exiv(const char *path, uint8_t **buffer, size_t *size, char **mime_type) {
      |                                                ^~~~~~~
```

**Comment by hamarituc** [comment]
> Does not compile with gcc-13 (default on ~amd64)

I cannot reproduce the error. I tried it on a clean stage 3 with `~amd64` keyword and GCC 13 and it compiles successfully. Can you please provide the output of `emerge --info` and the `build.log` file?

Please try again to compile on your setup with the attached patch. If it fixes the error, I will submit the patch upstream.

```diff
diff --git a/src/core/exif.h b/src/core/exif.h
index e95be4b6..c2966461 100644
--- a/src/core/exif.h
+++ b/src/core/exif.h
@@ -20,6 +20,8 @@
 #ifndef SRC_CORE_EXIF_H_
 #define SRC_CORE_EXIF_H_
 
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff --git a/src/io/SirilXISFReader.h b/src/io/SirilXISFReader.h
index c5c98740..d0ba3b19 100644
--- a/src/io/SirilXISFReader.h
+++ b/src/io/SirilXISFReader.h
@@ -20,6 +20,8 @@
 #ifndef SRC_IO_SIRILXISFREADER_H_
 #define SRC_IO_SIRILXISFREADER_H_
 
+#include <cstdint>
+
 extern "C" {
 #ifdef HAVE_CONFIG_H
 #include "config.h"
diff --git a/src/io/SirilXISFWraper.h b/src/io/SirilXISFWraper.h
index b280a07b..ea03e8aa 100644
--- a/src/io/SirilXISFWraper.h
+++ b/src/io/SirilXISFWraper.h
@@ -20,6 +20,8 @@
 #ifndef SRC_IO_SIRILXISFWRAPER_H_
 #define SRC_IO_SIRILXISFWRAPER_H_
 
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff --git a/src/io/avi_pipp/pipp_avi_write_dib.h b/src/io/avi_pipp/pipp_avi_write_dib.h
index 84a79652..caea754c 100644
--- a/src/io/avi_pipp/pipp_avi_write_dib.h
+++ b/src/io/avi_pipp/pipp_avi_write_dib.h
@@ -1,6 +1,8 @@
 #ifndef PIPP_AVI_WRITE_DIB_H
 #define PIPP_AVI_WRITE_DIB_H
 
+#include <cstdint>
+
 #include "pipp_video_write.h"
 #include "pipp_avi_write.h"
 
diff --git a/src/io/avi_pipp/pipp_buffer.h b/src/io/avi_pipp/pipp_buffer.h
index 53c54a68..c2cf7145 100644
--- a/src/io/avi_pipp/pipp_buffer.h
+++ b/src/io/avi_pipp/pipp_buffer.h
@@ -19,7 +19,6 @@
 #ifndef PIPP_BUFFER_H
 #define PIPP_BUFFER_H
 
-#include <cstdint>
 #include <cstdint>
 #include <cstddef>
 
```

**Comment by ceamac** [comment]
I think we have different sources (although the manifest is ok).

```
 * siril-1.2.0-rc1.tar.bz2 BLAKE2B SHA512 size ;-) ...                                                                                                                                                      [ ok ]
>>> Unpacking source...
>>> Unpacking siril-1.2.0-rc1.tar.bz2 to /mnt/tmp/portage/sci-astronomy/siril-1.2.0_rc1/work
>>> Source unpacked in /mnt/tmp/portage/sci-astronomy/siril-1.2.0_rc1/work
>>> Preparing source in /mnt/tmp/portage/sci-astronomy/siril-1.2.0_rc1/work/siril-1.2.0-rc1 ...
 * Applying siril-docfiles.patch ...                                                                                                                                                                        [ ok ]
 * Applying siril-1.2.0_rc1-fix-gcc-13.patch ...
patching file src/core/exif.h
can't find file to patch at input line 18
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/src/io/SirilXISFReader.h b/src/io/SirilXISFReader.h
|index c5c98740..d0ba3b19 100644
|--- a/src/io/SirilXISFReader.h
|+++ b/src/io/SirilXISFReader.h
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 31
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/src/io/SirilXISFWraper.h b/src/io/SirilXISFWraper.h
|index b280a07b..ea03e8aa 100644
|--- a/src/io/SirilXISFWraper.h
|+++ b/src/io/SirilXISFWraper.h
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
patching file src/io/avi_pipp/pipp_avi_write_dib.h
patching file src/io/avi_pipp/pipp_buffer.h
```

There is no `SirilXISFWraper.h` or `SirilXISFReader.h` in my sources.

```
viorel@gandalf ~/src/gentoo/sci-astronomy/siril $ ls -l /mnt/distfiles/siril-1.2.0-rc1.tar.bz2 
-rw-rw-r-- 1 portage portage 4353720 iun  9 11:48 /mnt/distfiles/siril-1.2.0-rc1.tar.bz2
viorel@gandalf ~/src/gentoo/sci-astronomy/siril $ sha256sum /mnt/distfiles/siril-1.2.0-rc1.tar.bz2
53098c88d1de235fbe27aa5362f4e94c3219574b3a503db426ae89017a8abaea  /mnt/distfiles/siril-1.2.0-rc1.tar.bz2
```

[build.log.txt](https://github.com/gentoo/gentoo/files/11714044/build.log.txt)
[emerge-info.txt](https://github.com/gentoo/gentoo/files/11714045/emerge-info.txt)

**Comment by hamarituc** [comment]
> I think we have different sources (although the manifest is ok).

Sorry I prepared the patch in the `master` branch and didn't noticed that there were already changes which needed to be backported. This patch should apply cleanly.

```diff
diff --git a/src/core/exif.h b/src/core/exif.h
index e95be4b6..c2966461 100644
--- a/src/core/exif.h
+++ b/src/core/exif.h
@@ -20,6 +20,8 @@
 #ifndef SRC_CORE_EXIF_H_
 #define SRC_CORE_EXIF_H_
 
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff --git a/src/io/avi_pipp/pipp_avi_write_dib.h b/src/io/avi_pipp/pipp_avi_write_dib.h
index 84a79652..caea754c 100644
--- a/src/io/avi_pipp/pipp_avi_write_dib.h
+++ b/src/io/avi_pipp/pipp_avi_write_dib.h
@@ -1,6 +1,8 @@
 #ifndef PIPP_AVI_WRITE_DIB_H
 #define PIPP_AVI_WRITE_DIB_H
 
+#include <cstdint>
+
 #include "pipp_video_write.h"
 #include "pipp_avi_write.h"
 
diff --git a/src/io/avi_pipp/pipp_buffer.h b/src/io/avi_pipp/pipp_buffer.h
index 53c54a68..c2cf7145 100644
--- a/src/io/avi_pipp/pipp_buffer.h
+++ b/src/io/avi_pipp/pipp_buffer.h
@@ -19,7 +19,6 @@
 #ifndef PIPP_BUFFER_H
 #define PIPP_BUFFER_H
 
-#include <cstdint>
 #include <cstdint>
 #include <cstddef>
 
```

**Comment by hamarituc** [comment]
It fails with GCC 13 and `exif` use flag disabled. I can confirm the patch fixes this issue.

**Comment by ceamac** [comment]
Now it doesn't link.  Missing dependency?

[build.log.txt.gz](https://github.com/gentoo/gentoo/files/11714142/build.log.txt.gz)

**Comment by hamarituc** [comment]
> Now it doesn't link. Missing dependency?
> 
> [build.log.txt.gz](https://github.com/gentoo/gentoo/files/11714142/build.log.txt.gz)

It was an optional dependency not treated as such at some points in the source code. The issue was fixed upstream: https://gitlab.com/free-astro/siril/-/issues/1129

File: sci-astronomy/siril/siril-1.2.0_rc1.ebuild [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	sci-libs/fftw:3.0=
+	sci-libs/gsl:=
+	x11-libs/cairo
+	>=x11-libs/gtk+-3.20.0:3
+	curl? ( net-misc/curl )
+	exif? ( >=media-gfx/exiv2-0.25 )
+	ffmpeg? ( media-video/ffmpeg:= )
+	heif? ( media-libs/libheif )
+	libconfig? ( >=dev-libs/libconfig-1.4[cxx] )
+	jpeg? ( media-libs/libjpeg-turbo:= )
```
Comment (ceamac) [inline sci-astronomy/siril/siril-1.2.0_rc1.ebuild:40]: Minor sorting issue, `jpeg` comes before `libconfig`

**Review by ceamac — APPROVED** [review]
Please check the output of `qa-vdb` from `iwdevtools`.

With default USE flags:
```
 * VDB: detected possibly incorrect RDEPEND (sci-astronomy/siril-1.2.0_rc1)
 * dev-libs/sleef:=          < 
 * media-libs/librtprocess:= | media-libs/librtprocess
 * sci-libs/cfitsio          | sci-libs/cfitsio:=
 *                           > x11-libs/gdk-pixbuf:2
 *                           > x11-libs/pango
```

The dependencies on the left or `<` may or may not be used (it only checks if they are directly linked, so they still mey be needed).  The dependencies on the right of `>` are linked to, so they should be added to RDEPEND.  The `|` is self-explanatory.

With all USE flags:
```
 * VDB: detected possibly incorrect RDEPEND (sci-astronomy/siril-1.2.0_rc1)
 * dev-libs/libconfig        | dev-libs/libconfig:=
 * dev-libs/sleef:=          < 
 * media-gfx/exiv2           | media-gfx/exiv2:=
 * media-libs/libheif        | media-libs/libheif:=
 * media-libs/libpng         | media-libs/libpng:=
 * media-libs/libraw         | media-libs/libraw:=
 * media-libs/librtprocess:= | media-libs/librtprocess
 * sci-astronomy/wcslib      | sci-astronomy/wcslib:=
 * sci-libs/cfitsio          | sci-libs/cfitsio:=
 *                           > x11-libs/gdk-pixbuf:2
 *                           > x11-libs/pango
```

**Pull request closed without merging**